### ISSUE TITLE
test01

### ISSUE THREAD

**Issue by ivanov-coffee** [ivanov-coffee/IvanovCoffee#1]
from transformers import CLIPProcessor, CLIPModel
from PIL import Image
import requests
import torch
# CLIPモデルをtransformersライブラリを用いてダウンロード
model = CLIPModel.from_pretrained("openai/clip-vit-base-patch32")
processor = CLIPProcessor.from_pretrained("openai/clip-vit-base-patch32")
# 画像の取得

url = "https://huggingface.co/datasets/huggingface/documentation-
images/resolve/main/cats.png"

image = Image.open(requests.get(url, stream=True).raw)
# 自然言語での候補を準備
texts = ["a photo of a cat", "a photo of a dog", "a photo of a robot"]
# 前処理と推論
inputs = processor(text=texts, images=image, return_tensors="pt",
padding=True)
with torch.no_grad():

outputs = model(**inputs)
logits_per_image = outputs.logits_per_image
probs = logits_per_image.softmax(dim=1)
# 結果の表示
for text, prob in zip(texts, probs[0]):
print(f"{text}: {prob.item():.4f}")

**Comment by ivanov-coffee** [comment]
import os
import torch
import torch.nn as nn
import torch.optim as optim
from torch.utils.data import Subset, DataLoader
from torchvision import datasets, transforms
import wandb
# MLPモデルの定義
class MLP(nn.Module):
def __init__(self):
super(MLP, self).__init__()
self.model = nn.Sequential(
nn.Flatten(),
nn.Linear(28 * 28, 256),
nn.ReLU(),
nn.Linear(256, 128),
nn.ReLU(),
nn.Linear(128, 10)
)
def forward(self, x):
return self.model(x)

config = {
"epochs": 5,
"batch_size": 64,
"learning_rate": 0.001,
"subset_size": 1000
}
# wandb初期化
wandb.init(project="mnist-mlp", name="subset-training", config=config)
# データ前処理
transform = transforms.Compose([transforms.ToTensor(),
transforms.Normalize((0.5,), (0.5,))])
train_dataset = datasets.MNIST(root="./data", train=True, download=True,
transform=transform)
test_dataset = datasets.MNIST(root="./data", train=False, download=True,
transform=transform)
# 学習データの一部のみ使用
subset_indices = list(range(wandb.config.subset_size))
train_dataset = Subset(train_dataset, subset_indices)
train_loader = DataLoader(train_dataset,
batch_size=wandb.config.batch_size, shuffle=True)
test_loader = DataLoader(test_dataset, batch_size=wandb.config.batch_size,
shuffle=False)
# モデル・損失関数・オプティマイザの定義
model = MLP()
criterion = nn.CrossEntropyLoss()
optimizer = optim.Adam(model.parameters(), lr=wandb.config.learning_rate)
epochs = wandb.config.epochs
for epoch in range(epochs):
total_loss = 0
# 学習ループ
model.train()
for images, labels in train_loader:
optimizer.zero_grad()
outputs = model(images)
loss = criterion(outputs, labels)
loss.backward()
optimizer.step()

total_loss += loss.item()
avg_loss = total_loss / len(train_loader)
# 評価ループ
model.eval()
correct = 0
total = 0
with torch.no_grad():
for images, labels in test_loader:
outputs = model(images)
_, predicted = torch.max(outputs, 1)
correct += (predicted == labels).sum().item()
total += labels.size(0)
test_accuracy = correct / total
print(f"Epoch [{epoch+1}/{epochs}], Loss: {avg_loss:.4f}, Test
Accuracy: {test_accuracy:.4f}")
# wandbでログを送信
wandb.log({
"epoch": epoch + 1,
"loss": avg_loss,
"test_accuracy": test_accuracy
})
# モデルの保存
os.makedirs("model", exist_ok=True)
torch.save(model.state_dict(), "./model/mnist_mlp_model.pt")
wandb.finish()

**Comment by ivanov-coffee** [comment]
import pybullet as p
import time
import pybullet_data
physicsClient = p.connect(p.GUI)#or p.DIRECT for non-graphical version

p.setAdditionalSearchPath(pybullet_data.getDataPath()) #optionally
p.setGravity(0,0,-10)
planeId = p.loadURDF("plane.urdf")
startPos = [0,0,1]
startOrientation = p.getQuaternionFromEuler([0,0,0])
boxId = p.loadURDF("r2d2.urdf",startPos, startOrientation)
#set the center of mass frame (loadURDF sets base link frame)
startPos/Ornp.resetBasePositionAndOrientation(boxId, startPos,
startOrientation)
for i in range (10000):
p.stepSimulation()
time.sleep(1./240.)
cubePos, cubeOrn = p.getBasePositionAndOrientation(boxId)
print(cubePos,cubeOrn)
p.disconnect()





import pybullet as p
import time
import pybullet_data
import numpy as np
from PIL import Image
physicsClient = p.connect(p.GUI)
p.setAdditionalSearchPath(pybullet_data.getDataPath())
p.setGravity(0, 0, -10)
planeId = p.loadURDF("plane.urdf")
startPos = [0, 0, 1]
startOrientation = p.getQuaternionFromEuler([0, 0, 0])
boxId = p.loadURDF("r2d2.urdf", startPos, startOrientation)
# シミュレーションステップ
for i in range(240):
p.stepSimulation()
time.sleep(1. / 240.)
# カメラの位置と向きの設定
viewMatrix = p.computeViewMatrix(
cameraEyePosition=[1, 1, 1],
cameraTargetPosition=[0, 0, 0],
cameraUpVector=[0, 0, 1]
)
# カメラのビューの設定
projectionMatrix = p.computeProjectionMatrixFOV(
fov=60, aspect=1.0, nearVal=0.1, farVal=100.0
)

# カメラ画像を取得
width, height, rgbImg, depthImg, segImg = p.getCameraImage(
width=512,
height=512,
viewMatrix=viewMatrix,
projectionMatrix=projectionMatrix
)
img = Image.fromarray(rgbImg)
img.save("screenshot.png")
p.disconnect()
